codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/handler/SockJsHandler.java

```diff
@@ -0,0 +1,273 @@
+     */
+    public SockJsHandler(final SockJsServiceFactory... factories) {
+        checkNotNullAndNotEmpty(factories, "factories");
+        for (SockJsServiceFactory factory : factories) {
+            this.factories.put(factory.config().prefix(), factory);
+        }
+    }
+
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        if (msg instanceof FullHttpRequest) {
+            final FullHttpRequest request = (FullHttpRequest) msg;
+            final String path = new QueryStringDecoder(request.uri()).path();
+            for (SockJsServiceFactory factory : factories.values()) {
+                if (path.startsWith(factory.config().prefix())) {
+                    handleService(factory, request, ctx);
+                    return;
+                }
+            }
+            writeNotFoundResponse(request, ctx);
+        } else {
+            ctx.fireChannelRead(msg);
+        }
+    }
+
+    private static void handleService(final SockJsServiceFactory factory,
+                                      final FullHttpRequest request,
+                                      final ChannelHandlerContext ctx) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("RequestUri : [{}]", request.uri());
+        }
+        final SockJsConfig sockJsConfig = factory.config();
+        final String path = pathWithoutPrefix(sockJsConfig.prefix(), request);
+        if (Greeting.matches(path)) {
+            writeResponse(ctx.channel(), request, Greeting.response(request));
+            return;
+        }
+        if (Info.matches(path)) {
+            writeResponse(ctx.channel(), request, Info.response(sockJsConfig, request));
+            return;
+        }
+        if (Iframe.matches(path)) {
+            writeResponse(ctx.channel(), request, Iframe.response(sockJsConfig, request));
+            return;
+        }
+        if (RawWebSocketTransport.matches(path)) {
+            addTransportHandler(new RawWebSocketTransport(sockJsConfig, factory.newService()), ctx);
+            ctx.fireChannelRead(request);
+            return;
+        }
+        final SockJsPath sockJsPath = SockJsPaths.parse(path);
+        if (sockJsPath.isValid()) {
+            handleSession(factory, sockJsConfig, request, ctx, sockJsPath.transport(), sockJsPath.sessionId());
+            return;
+        }
+        writeNotFoundResponse(request, ctx);
+    }
+
+    private static void handleSession(final SockJsServiceFactory factory,
+                                      final SockJsConfig sockJsConfig,
+                                      final FullHttpRequest request,
+                                      final ChannelHandlerContext ctx,
+                                      final TransportType transportType,
+                                      final String sessionId) throws Exception {
+        switch (transportType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)